src/lib.rs

```diff
@@ -384,6 +382,11 @@ impl Navigator {
         self.led.set_led_all(false);
     }
 
+    /// Initialize the log service. Which can display feedback for some processes.
+    pub fn init_logger(&mut self) {
```

Comment:
This is not a mutt self and also is not returning.

Reply:
Actually it works like follows:
```
let mut nav = Navigator::new();
nav.init_logger();
nav.init();
nav.set_pwm_freq_hz(60.0);
```

You suggest to:
use a separate `Navigator::init_logger()` 
 
 ```
let mut nav = Navigator::new();
Navigator::init_logger();
nav.init();
nav.set_pwm_freq_hz(60.0);
```

Reply:
oh, just &self.
working on it.

raspberry-pi
Creating your navigator module!
Setting up your navigator, ahoy!
[2023-09-14T15:10:56Z INFO  navigator_rs] Logger initialized successfully
[2023-09-14T15:10:57Z INFO  navigator_rs] PWM frequency set to 60.00 Hz. Prescaler value: 99


and

raspberry-pi
Creating your navigator module!
Setting up your navigator, ahoy!
[2023-09-14T15:10:56Z INFO  navigator_rs] Logger initialized successfully
[2023-09-14T15:11:45Z ERROR navigator_rs] Failed to initialize logger: attempted to set a logger after the logging system was already initialized.
[2023-09-14T15:10:57Z INFO  navigator_rs] PWM frequency set to 60.00 Hz. Prescaler value: 99